spec/features/customers/show_spec.rb

```diff
@@ -0,0 +1,58 @@
+require "rails_helper"
+
+RSpec.describe "Customer Show Page", type: :feature do
+    before(:each) do 
+        @supermarket1 = Supermarket.create!(name: "Corner Market", location: "123 Food St.")
+        @supermarket2 = Supermarket.create!(name: "Mimi Market", location: "456 Yum St.")
+
+        @customer1 = Customer.create!(name: "Sally Shopper")
+        @customer2 = Customer.create!(name: "Molly Spender")
+
+        @item1 = @supermarket1.items.create!(name: "Apple", price: 2)
+        @item2 = @supermarket1.items.create!(name: "Banana", price: 1)
+        @item3 = @supermarket2.items.create!(name: "Bread", price: 3)
+        
+        @customer1.items << @item1
+        @customer1.items << @item2
+    end
+
+    # 1 As a visitor, 
+    it 'shows customer name and items' do
+
+        # When I visit a customer show page,
+        visit "/customers/#{@customer1.id}"
+        # I see the customer's name,
+        expect(page).to have_content(@customer1.name)
+        # And I see its a list of its items
+        # including the item's name, price, and the name of the supermarket that it belongs to.
+        expect(page).to have_content(@item1.name)
+        expect(page).to have_content(@item1.price)
```

Comment:
Consider anchoring these assertions to the text label they will appear next to on the page. In order to prevent any false positives, you can assert `expect(page).to have_content("Price: ${@item1.price}")`

Reply:
Especially when you are just asserting the presence of a number on the page, you can get false positives if there are numbers elsewhere on the page (like a phone number in the footer) so it's a good idea to anchor the data next to its label